docs/sources/developers/plugins/add-anonymous-usage-reporting.md

```diff
@@ -0,0 +1,152 @@
+---
+aliases:
+  - /docs/grafana/latest/developers/plugins/add-anonymous-usage-reporting/
+title: Add anonymous usage reporting
+---
+
+# Add anonymous usage reporting to you plugin
+
+The Grafana server administrator has the possiblity to configure [anonymous usage tracking]({{< relref "../../setup-grafana/configure-grafana/#reporting_enabled" >}}).
+
+By adding usage tracking to your plugin you will send events of how your plugin is being used to the configured tracking system.
+
+Lets say we have a QueryEditor that looks something like the example below. It it has an editor field where you can write your query and a query type selector so you can select what kind of query result you are expecting that query to return.
+
+```ts
+import React, { ReactElement } from 'react';
+import { InlineFieldRow, InlineField, Select, CodeEditor } from '@grafana/ui';
+import type { EditorProps } from './types';
+
+export function QueryEditor(props: EditorProps): ReactElement {
+  const { datasource, query, onChange, onRunQuery } = props;
+  const queryType = { value: query.value ?? 'timeserie' };
+  const queryTypes = [
+    {
+      label: 'Timeserie',
+      value: 'timeserie',
+    },
+    {
+      label: 'Table',
+      value: 'table',
+    },
+  ];
+
+  const onChangeQueryType = (type: string) => {
+    onChange({
+      ...query,
+      queryType: type,
+    });
+    runQuery();
+  };
+
+  const onChangeRawQuery = (rawQuery: string) => {
+    onChange({
+      ...query,
+      rawQuery: type,
+    });
+    runQuery();
+  };
+
+  return (
+    <>
+      <div>
+        <CodeEditor
+          height="200px"
+          showLineNumbers={true}
+          language="sql"
+          onBlur={onChangeRawQuery}
+          value={query.rawQuery}
+        />
+      </div>
+      <InlineFieldRow>
+        <InlineField label="Query type" grow>
+          <Select options={queryTypes} onChange={onChangeQueryType} value={queryType} />
+        </InlineField>
+      </InlineFieldRow>
+    </>
+  );
+}
+```
+
+Lets say that we would like to track how the usage looks between time series and table queries. All you need to do is to add the `usePluginInteractionReporter` to fetch a report function which takes two arguments. The first one is the event name which is used to identify the interaction being made. It need to start with `grafana_plugin_` which makes it easier to differentiate plugin events from Grafana core events. The second argument is optional and should be used to attach contextual data to the event. In our example, that would be the query type. It is optional because it does not make sense to pass contextual data for all user interactions.
```

Comment:
I know it doesn't tie closely to the documentation part, but in case the event name _"need to start with `grafana_plugin_`"_, would it make sense to automatically prefix the event names with `grafana_plugin_` instead of [erroring out](https://github.com/grafana/grafana/blob/b804b2f073669e28827a18bfddc2a690f7a7dbec/packages/grafana-runtime/src/analytics/plugins/usePluginInteractionReporter.ts#L28)? In this case we could document that all events using this reporter would be prefixed. Or maybe that would be less explicit?

Reply:
I'm not sure. Then I guess we need to check that the name does not have `grafana_plugin_` in the beginning to prevent it from being `grafana_plugin_grafana_plugin_event_name`.

I don't have any strong opinions but I was moving back and forth between this and ended in letting the consumer state the whole thing to basically have the same value in the code that you would search/aggregate on in your analytics system.